pkg/core/consensus/blockgenerator/candidate/blockgenerator.go

```diff
@@ -47,14 +42,21 @@ type Generator interface {
 type generator struct {
 	*consensus.Emitter
 
-	executeFn consensus.ExecuteTxsFunc
+	callTimeout time.Duration
+	executeFn   consensus.ExecuteTxsFunc
 }
 
 // New creates a new block generator.
 func New(e *consensus.Emitter, executeFn consensus.ExecuteTxsFunc) Generator {
+	ct := config.Get().Timeout.TimeoutGetMempoolTXsBySize
+	if ct == 0 {
+		ct = 5
+	}
+
 	return &generator{
-		Emitter:   e,
-		executeFn: executeFn,
+		Emitter:     e,
+		executeFn:   executeFn,
+		callTimeout: time.Duration(ct) * time.Second,
```

Comment:
Not sure why you think this has been introduced now.  The rpcbus call timeout is there from day one. If you read the PR in depth you'll notice that this a minor change to avoid redundant timeout calculation on each rpcbus call.